subprojects/core/src/main/java/org/gradle/configuration/DefaultScriptPluginFactory.java

```diff
@@ -259,4 +261,19 @@ private ScriptTarget wrap(Object target, boolean isInitialPass) {
             }
         }
     }
+
+    // TODO:DAZ This is not nice: work out a better way to collect the plugin requests from invoking the plugins block.
+    // It would also be good to provide the `pluginsBlockLineNumber` prior to invocation.
+    private PluginRequests getInitialPluginRequests(ScriptRunner<? extends BasicScript, ?> initialRunner) {
+        if (initialRunner.getRunDoesSomething()) {
+            BasicScript script = initialRunner.getScript();
+            if (script instanceof ProjectScript) {
```

Comment:
This needs to be generic enough to support `Init` and `Settings` moving forward too. Is this possible to do easily?

Reply:
I'm sure it's possible, not sure how "easy". In the end, I think this will require some further design to make it work well in different contexts.